### DISCUSSION TITLE
I Cant Seem To Get Sample Blocksize and custom slider values right for FLAC conversion (quality adjustment)

### DISCUSSION THREAD

**Discussion by impotentfield** [enzo1982/freac#445] [Audio Format Discussion]
in my experience, it seems that the quality of the flac file resides, where my walkman plays it, in the adjustment of sample blocksize. no matter the idea in apodization which seems to play a big role in my experience adjusting and listening, the sample blocksize adjustment, between multiple conversions seems to play a considerable role in the acoustic experience and the output of power in the headphones while listening (to mainly the acoustics). when i say output power i mean anything along the line of which i have been taking up looking at comments and opinions  - db, dynamic range, loudness. i cant seem to get the sample blocksize and also slider values in the customization just right where the, db, dynamic range, loudness, overall acoustic quality, play a role after the adjustment and conversion. as i have been adjusting and listening over time, these two factors seem to be the missing piece to my woes where i cant seem to get the right adjustment. can anyone help me with this, my flac files are becoming a disaster.

**Comment by Lee-Carre** [comment]
Hmm. Firstly, FLAC is lossless; the decoded waveform matches the input waveform, unless something weird is going on.

The differences you perceive, unless they're just perceptions via cognitive bias (such as confirmation bias), might be due to your playback device being portable, and thus power-constrained. I would suggest investing in a (portable) headphone amplifier, to achieve more consistent playback performance. This will more likely mean that your earphones become the limiting factor, though.

As for optimising variable sample block-lengths, I recently happened upon [Benchmark of Nayuki’s FLAC encoder](//www.nayuki.io/page/benchmark-of-nayukis-flac-encoder):

> The main feature I wanted to explore in my encoder was the use of variable block sizes. Xiph.Org’s documentation states in multiple places[0][1] that the reference encoder only implements logic to generate fixed/constant block sizes[2]. What I knew was that when given a set of possible block sizes to choose from (e.g. {2048, 3072, 4096, 6144} samples), it is possible to apply dynamic programming to find the optimal sequence of block sizes to use to encode the whole audio file.

Sounds to me like it's related to what you're trying to do.

If the output of your player is, somehow, dependent upon the encoding of the FLAC file, then it would seem that you need a more automated/systematic way of testing, to determine which blocksize yields the best results.
Some shell-scripting, to multiply-encode the same source to multiple different FLAC outputs, to then perform (preferably blinded) listening tests, would help. Assuming it's not a hardware problem, of course.
